main.tf

```diff
@@ -16,20 +16,40 @@ provider "aws" {
 }
 
 # ec2 instances
-resource "aws_instance" "node_server" {
-  count = (var.DOCKERIZED == "yes") ? 1 : (var.VALIDATOR_COUNT + var.SENTRY_COUNT + var.ARCHIVE_COUNT)
+resource "aws_instance" "bor_node_server" {
+  count = (var.DOCKERIZED == "yes") ? 1 : (var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT + var.BOR_ARCHIVE_COUNT)
   ami                    = var.INSTANCE_AMI
-  instance_type          = (count.index >= var.VALIDATOR_COUNT + var.SENTRY_COUNT) ? var.ARCHIVE_INSTANCE_TYPE: var.INSTANCE_TYPE
+  instance_type          = (count.index >= var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT) ? var.BOR_ARCHIVE_INSTANCE_TYPE: var.BOR_INSTANCE_TYPE
   key_name               = var.PEM_FILE
   vpc_security_group_ids = [aws_security_group.internet_facing_load_balancer_sg.id]
   subnet_id              = aws_subnet.devnet_public_subnet.id
 
   # instances' disks
   ebs_block_device {
     device_name = "/dev/sda1"
-    volume_size = (count.index >= var.VALIDATOR_COUNT + var.SENTRY_COUNT) ? var.ARCHIVE_DISK_SIZE_GB : var.DISK_SIZE_GB
-    volume_type = (count.index >= var.VALIDATOR_COUNT + var.SENTRY_COUNT) ? var.ARCHIVE_VOLUME_TYPE : var.VOLUME_TYPE
-    iops = (count.index >= var.VALIDATOR_COUNT + var.SENTRY_COUNT ) ? var.ARCHIVE_IOPS : var.IOPS
+    volume_size = (count.index >= var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT) ? var.BOR_ARCHIVE_DISK_SIZE_GB : var.BOR_DISK_SIZE_GB
+    volume_type = (count.index >= var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT) ? var.BOR_ARCHIVE_VOLUME_TYPE : var.BOR_VOLUME_TYPE
+    iops = (count.index >= var.BOR_VALIDATOR_COUNT + var.BOR_SENTRY_COUNT ) ? var.BOR_ARCHIVE_IOPS : var.BOR_IOPS
+  }
+
+  tags = {
+    Name = "${var.VM_NAME}_${count.index + 1}"
+  }
+}
+resource "aws_instance" "erigon_node_server" {
+  count = (var.DOCKERIZED == "yes") ? 1 : (var.ERIGON_VALIDATOR_COUNT + var.ERIGON_SENTRY_COUNT + var.ERIGON_ARCHIVE_COUNT)
```

Comment:
Yeah I see your point. I actually targeted the remote setup but ended up adding the condition for docker as well; it's probably not needed in case of Erigon. Let me know your thoughts.

Reply:
If we remove it, then we can never run a dockerized env with erigon only
But if we leave it, we'll create 2 servers which is an error. Maybe we need to move/rethink this check

Reply:
Can we have a separate resource block to handle dockerized cases like we are currently handling bor and erigon in separate resource blocks so does a third resource for docker makes sense?

Reply:
Done here: https://github.com/maticnetwork/matic-cli/pull/184/commits/858c3f213f94d97deaa9cfaf9cef0c0da661f7e7